test/test2.js

```diff
@@ -121,13 +138,34 @@ describe('Workflow Subdomains and Prefixes', function() {
     });
   });
 
-  it('can find a jointly-determined locale via middleware', function(done) {
+  it('can find a jointly-determined locale via middleware - case 1', function(done) {
     tryMiddleware('http://example.com/es', function(req) {
       assert(req.locale === 'us-es');
       done();
     });
   });
 
+  it('can find a jointly-determined locale via middleware - case 2', function (done) {
+    tryMiddleware('http://example.es/co', function (req) {
+      assert(req.locale === 'es-CO');
+      done();
+    });
+  });
+
+  it('can detect a root-level locale via middleware - case 1', function(done) {
+    tryMiddleware('http://example.com', function(req) {
+      assert(req.locale === 'us-en');
```

Comment:
Looks like you don't have a test for a non-root page in this situation.

Reply:
As part of my update, I've added a couple of tests to cover the cases you've mentioned.